sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -153,29 +162,43 @@ private boolean onFirstTransactionStarted(final @NotNull ITransaction transactio
     if (startData == null) {
       return false;
     }
-    transactionStartNanos = startData.startNanos;
+    profileStartNanos = startData.startNanos;
     profileStartCpuMillis = startData.startCpuMillis;
-
-    currentTransaction = transaction;
-
-    currentProfilingTransactionData =
-        new ProfilingTransactionData(transaction, transactionStartNanos, profileStartCpuMillis);
     return true;
   }
 
+  @Override
+  public synchronized void bindTransaction(final @NotNull ITransaction transaction) {
+    // If the profiler is running, but no profilingTransactionData is set, we bind it here
+    if (transactionsCounter > 0 && currentProfilingTransactionData == null) {
```

Comment:
nit: Given that we're anyway inside `bindTransaction` already, do we need to check the `transactionsCounter` here?

Reply:
yep, that `transactionsCounter` checks if the profiler is running.
Without it, you could bind the profiler to a transaction even if it's not running. In that case, `transactionsCounter` wouldn't be reset, so when the profiler starts for another transaction, it would keep the reference to the old one